rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -220,11 +238,18 @@ unsafe fn update_refs<SetHp: Fn(u32)>(set_hp: SetHp, heap_base: u32) {
 
 /// Thread forward pointers in object
 unsafe fn thread_fwd_pointers(obj: *mut Obj, heap_base: u32) {
-    visit_pointer_fields(obj, obj.tag(), heap_base as usize, |field_addr| {
-        if (*field_addr).get_ptr() > obj as usize {
-            thread(field_addr)
-        }
-    });
+    visit_pointer_fields(
+        &mut (),
```

Comment:
What does this do? Is that a dummy pointer?

Reply:
Yes. The receiving side wants a `*mut`, so we give it one.